webapp/src/components/assetEditor/editor.tsx

```diff
@@ -112,7 +120,35 @@ export class AssetEditor extends Editor {
                 break;
         }
 
-        fieldView.onHide(() => cb(fieldView.getResult()));
+        fieldView.onHide(() => {
+            const result = fieldView.getResult();
+            if (asset.type == pxt.AssetType.Tilemap) result.data = this.updateTilemapTiles(result.data);
+            cb(result);
+        });
         fieldView.show();
     }
+
+    protected updateTilemapTiles = (data: pxt.sprite.TilemapData): pxt.sprite.TilemapData => {
+        const project = pxt.react.getTilemapProject();
```

Comment:
move update of deleted and edited tiles out of individual "create"/"edit" callbacks, always update all tiles and trim tileset after exiting field editor